.test-infra/dataproc/create_flink_cluster.sh

```diff
@@ -0,0 +1,144 @@
+#!/usr/bin/env bash
+#    Licensed to the Apache Software Foundation (ASF) under one or more
+#    contributor license agreements.  See the NOTICE file distributed with
+#    this work for additional information regarding copyright ownership.
+#    The ASF licenses this file to You under the Apache License, Version 2.0
+#    (the "License"); you may not use this file except in compliance with
+#    the License.  You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+#
+#    Runs init actions for Docker, Portability framework (Beam) and Flink cluster
+#    and opens an SSH tunnel to connect with Flink easily and run Beam jobs.
+#
+#    Pass the following command line args to run this script:
+#    $1: Cluster name
+#    $2: GCS bucket url for Dataproc resources (init actions)
+#    $3: SDK Harness' images repository from which images will be downloaded
+#    $4: SDK Harness' images to pull on dataproc workers (python,java,go)
+#    $5: Url to Flink .tar archive to be installed on the cluster
+#    $6: Number of Flink workers
+#    $7: Number of Flink slots
+#    $8: Detached mode: should the SSH tunnel run in detached mode?
+#
+#    Example usage:
+#
+#    ./create_flink_cluster.sh \
+#    flink \
+#    gs://<GCS_BUCKET>/flink \
+#    gcr.io/<IMAGE_REPOSITORY_PATH> \
+#    python,go,java \
+#    http://archive.apache.org/dist/flink/flink-1.7.0/flink-1.7.0-bin-hadoop28-scala_2.12.tgz \
+#    2 \
+#    1 \
+#    false
+#
+set -Eeuxo pipefail
+
+DATAPROC_VERSION=1.2
+
+CLUSTER_NAME=$1
+MASTER_NAME="$CLUSTER_NAME-m"
+
+# GCS properties
+GCS_BUCKET=$2
+INIT_ACTIONS_FOLDER_NAME="init-actions"
+FLINK_INIT="$GCS_BUCKET/$INIT_ACTIONS_FOLDER_NAME/flink.sh"
+BEAM_INIT="$GCS_BUCKET/$INIT_ACTIONS_FOLDER_NAME/beam.sh"
+DOCKER_INIT="$GCS_BUCKET/$INIT_ACTIONS_FOLDER_NAME/docker.sh"
+
+# Portability options
+HARNESS_IMAGES_REPOSITORY_URL=$3
+HARNESS_IMAGES_TO_PULL=$4
+
+# Flink properties
+FLINK_DOWNLOAD_URL=$5
+FLINK_LOCAL_PORT=8081
+FLINK_NUM_WORKERS=$6
+TASK_MANAGER_SLOTS=$7
+TASK_MANAGER_MEM=10240
+YARN_APPLICATION_MASTER=""
+
+DETACHED_MODE=$8
+
+function upload_init_actions() {
+  echo "Uploading initialization actions to GCS bucket: $GCS_BUCKET"
```

Comment:
Uploading init action files to private gcs bucket.

Reply:
Yup, I see : )